fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -365,7 +365,8 @@ class ApplyMetadataMutationsImpl {
 			if ((!m.param1.startsWith(excludedServersPrefix) && m.param1 != excludedServersVersionKey) &&
 			    (!m.param1.startsWith(failedServersPrefix) && m.param1 != failedServersVersionKey) &&
 			    (!m.param1.startsWith(excludedLocalityPrefix) && m.param1 != excludedLocalityVersionKey) &&
-			    (!m.param1.startsWith(failedLocalityPrefix) && m.param1 != failedLocalityVersionKey)) {
+			    (!m.param1.startsWith(failedLocalityPrefix) && m.param1 != failedLocalityVersionKey) &&
+			    !m.param1.startsWith(storageEngineParamsPrefix) && !m.param1.startsWith(tssStorageEngineParamsPrefix)) {
```

Comment:
Why does changing the storage engine params require a recovery?  If it is to force DD to restart, it is possible to do that separately if the transaction that sets the db config can also update the MoveKeysLock or simply call ```DDConfiguration().trigger.update(tr))`.

Reply:
It does not need a recovery. The code added here to skip the recovery if the changed `\xff/conf/...` keys are in these ranges.